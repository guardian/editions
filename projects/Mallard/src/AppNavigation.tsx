import {
	CardStyleInterpolators,
	createStackNavigator,
} from '@react-navigation/stack';
import React from 'react';
import { Animated } from 'react-native';
import type { RootStackParamList } from './navigation/NavigationModels';
import { RouteNames } from './navigation/NavigationModels';
import { ArticleWrapper } from './navigation/navigators/article';
import { EditionsMenuScreen } from './screens/editions-menu-screen';
import { HomeScreen } from './screens/home-screen';
import { AuthSwitcherScreen } from './screens/identity-login-screen';
import { IssueScreen } from './screens/issue-screen';
import { OnboardingConsentScreen } from './screens/onboarding-screen';
import { SettingsScreen } from './screens/settings-screen';
import { AlreadySubscribedScreen } from './screens/settings/already-subscribed-screen';
import { ApiScreen } from './screens/settings/api-screen';
import { BetaProgrammeFAQsScreen } from './screens/settings/beta-programme-faqs';
import { CasSignInScreen } from './screens/settings/cas-sign-in-screen';
import { CreditsScreen } from './screens/settings/credits-screen';
import { EditionsScreen } from './screens/settings/editions-screen';
import { FAQScreen } from './screens/settings/faq-screen';
import {
	GdprConsentScreen,
	GdprConsentScreenForOnboarding,
} from './screens/settings/gdpr-consent-screen';
import { HelpScreen } from './screens/settings/help-screen';
import { ManageEditionsScreenWithHeader } from './screens/settings/manage-editions-screen';
import {
	PrivacyPolicyScreen,
	PrivacyPolicyScreenForOnboarding,
} from './screens/settings/privacy-policy-screen';
import { SubscriptionDetailsScreen } from './screens/settings/subscription-details-screen';
import { TermsAndConditionsScreen } from './screens/settings/terms-and-conditions-screen';

const Stack = createStackNavigator<RootStackParamList>();

const { multiply } = Animated;

const cardStyleInterpolator = (props: any) => {
	const translateX = multiply(
		props.current.progress.interpolate({
			inputRange: [0, 1],
			outputRange: [props.layouts.screen.width, 0],
			extrapolate: 'clamp',
		}),
		props.inverted,
	);

	return {
		// ...CardStyleInterpolators.forHorizontalIOS(props),
		cardStyle: {
			overflow: 'hidden',
			transform: [
				// Translation for the animation of the current card
				{
					translateX,
				},
			],
		},
		overlayStyle: {
			opacity: props.current.progress.interpolate({
				inputRange: [0, 1],
				outputRange: [0, 0.5],
				extrapolate: 'clamp',
			}),
		},
	};
};

const RootStack = () => {
	return (
		<Stack.Navigator
			initialRouteName={RouteNames.Home}
			screenOptions={{ gestureEnabled: false, headerShown: false }}
		>
			<Stack.Screen
				name={RouteNames.Issue}
				component={IssueScreen}
				options={{}}
			/>
			<Stack.Screen
				name={RouteNames.IssueList}
				component={HomeScreen}
				options={{
					gestureDirection: 'horizontal',
					cardStyle: { backgroundColor: 'transparent' },
					cardOverlayEnabled: true,
					cardStyleInterpolator,
				}}
			/>
			<Stack.Screen
				name={RouteNames.EditionsMenu}
				component={EditionsMenuScreen}
				options={{
					gestureDirection: 'horizontal-inverted',
					cardStyle: { backgroundColor: 'transparent' },
					cardOverlayEnabled: true,
					cardStyleInterpolator,
				}}
			/>
			<Stack.Screen
				name={RouteNames.Article}
				component={ArticleWrapper}
				options={{
					cardStyleInterpolator:
						CardStyleInterpolators.forModalPresentationIOS,
					gestureEnabled: true,
					gestureDirection: 'vertical',
				}}
			/>
			{/* <Stack.Screen
                name={RouteNames.Article}
                component={SlideCardJames}
                options={{
                    cardStyleInterpolator: props => {
                        return {
                            ...CardStyleInterpolators.forModalPresentationIOS(
                                props,
                            ),
                        }
                    },
                }}
            /> */}
			<Stack.Screen
				name={RouteNames.Settings}
				component={SettingsScreen}
			/>
			<Stack.Screen
				name={RouteNames.SignIn}
				component={AuthSwitcherScreen}
			/>
			{/* <Stack.Screen
                name={RouteNames.OnboardingStart}
            /> */}
			<Stack.Screen
				name={RouteNames.OnboardingConsent}
				component={OnboardingConsentScreen}
			/>
			<Stack.Screen
				name={RouteNames.OnboardingConsentInline}
				component={GdprConsentScreenForOnboarding}
			/>
			<Stack.Screen
				name={RouteNames.PrivacyPolicyInline}
				component={PrivacyPolicyScreenForOnboarding}
			/>
			<Stack.Screen
				name={RouteNames.AlreadySubscribed}
				component={AlreadySubscribedScreen}
			/>
			<Stack.Screen
				name={RouteNames.CasSignIn}
				component={CasSignInScreen}
			/>
			<Stack.Screen
				name={RouteNames.ManageEditionsSettings}
				component={ManageEditionsScreenWithHeader}
			/>
			{/** @TODO Fix the enable all button */}
			<Stack.Screen
				name={RouteNames.GdprConsent}
				component={GdprConsentScreen}
			/>
			<Stack.Screen
				name={RouteNames.PrivacyPolicy}
				component={PrivacyPolicyScreen}
			/>

			{/* ==== Inspect from here === */}
			<Stack.Screen
				name={RouteNames.Edition}
				component={EditionsScreen}
			/>
			<Stack.Screen
				name={RouteNames.TermsAndConditions}
				component={TermsAndConditionsScreen}
			/>
			<Stack.Screen
				name={RouteNames.BetaProgrammeFAQs}
				component={BetaProgrammeFAQsScreen}
			/>
			<Stack.Screen name={RouteNames.Help} component={HelpScreen} />
			<Stack.Screen name={RouteNames.Credits} component={CreditsScreen} />
			<Stack.Screen name={RouteNames.FAQ} component={FAQScreen} />
			<Stack.Screen
				name={RouteNames.SubscriptionDetails}
				component={SubscriptionDetailsScreen}
			/>
			<Stack.Screen name={RouteNames.Endpoints} component={ApiScreen} />
			{/* Turned off to remove Promise rejection error on Android */}
			{/* <Stack.Screen
                name={RouteNames.Storybook}
                component={StorybookScreen}
            /> */}
		</Stack.Navigator>
	);
};

export { RootStack };
