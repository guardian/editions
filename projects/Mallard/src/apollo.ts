/**
 * Setup Apollo framework. At the moment we use Apollo Local State management
 * mechanism and don't fetch data from GraphQL endpoints.
 */
import { ApolloClient } from 'apollo-client'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { SETTINGS_RESOLVERS } from './helpers/settings/resolvers'
import { resolveWeather } from './helpers/location'

/**
 * Resolvers is what Apollo uses to get the value of field that has never been
 * cached before. For example if you have a query `{ foo @client }`, Apollo
 * looks for the field `foo` at the root of the cache. If not available, it'll
 * look for the resolver `Query.foo` (`Query` is just the "namespace" where
 * all queries start).
 *  *
 * Resolvers are functions, which can be asynchonous. They can gather data from
 * local storage (ex. settings), from local APIs (ex. permissions, geolocation),
 * from remote APIs (ex. REST API).
 *
 * Once a resolver returns a value, Apollo will refresh automatically
 * any component that was requiring this in a query via the `useQuery` function.
 */
const RESOLVERS = {
    Query: {
        ...SETTINGS_RESOLVERS,
        weather: resolveWeather,
    },
}

/**
 * The `Link` is the interface Apollo uses to reach GraphQL servers. Since we
 * don't fetch anything from GraphQL endpoints at the moment, we use a no-op
 * `Link` object. We *must* throw to prevent bad queries from carrying through.
 *
 * Apollo would try to use the `Link` if you have a query such as `{ foo }`
 * without the `@client` annotation. In our current use case, `@client` always
 * needs to be specified.
 */
const link = {
    split: () => {
        throw new Error('Not implemented. Did you forget to use `@client`?')
    },
    concat: () => {
        throw new Error('not implemented Did you forget to use `@client`?')
    },
    request: () => {
        throw new Error('not implemented Did you forget to use `@client`?')
    },
}

export const createApolloClient = () =>
    new ApolloClient({
        cache: new InMemoryCache(),
        link,
        resolvers: RESOLVERS,
    })
