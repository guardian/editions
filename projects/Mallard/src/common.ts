export * from '../../Apps/common/src/index';
