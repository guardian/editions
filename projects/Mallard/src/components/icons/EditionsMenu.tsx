import React from 'react';
import Svg, { Circle } from 'react-native-svg';

const EditionsMenu = () => (
	<Svg width="40" height="40" viewBox="0 0 42 42" fill="none">
		<Circle cx="21" cy="21" r="20.5" fill="#052962" stroke="white" />
		<Circle cx="14.1751" cy="13.125" r="6.825" fill="white" />
		<Circle cx="14.1751" cy="27.825" r="6.825" fill="#FF7F0F" />
		<Circle cx="28.875" cy="13.125" r="6.825" fill="#FFBAC8" />
		<Circle cx="28.875" cy="27.825" r="6.825" fill="#90DCFF" />
	</Svg>
);

export { EditionsMenu };
