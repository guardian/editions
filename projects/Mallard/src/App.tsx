// Welcome to the main entry point of the app.
//
// In this file, we'll be kicking off our app or storybook.

import { ApolloProvider } from '@apollo/react-hooks';
import AsyncStorage from '@react-native-community/async-storage';
import type ApolloClient from 'apollo-client';
import React from 'react';
import { AppState, StatusBar, StyleSheet, View } from 'react-native';
import SplashScreen from 'react-native-splash-screen';
import { eventEmitter } from 'src/helpers/event-emitter';
import { weatherHider } from 'src/helpers/weather-hider';
import {
	ConfigProvider,
	largeDeviceMemory,
} from 'src/hooks/use-config-provider';
import { NavPositionProvider } from 'src/hooks/use-nav-position';
import { setUserId } from 'src/services/ophan';
import { AppNavigation } from './AppNavigation';
import { AccessProvider } from './authentication/AccessContext';
import type { IdentityAuthData } from './authentication/authorizers/IdentityAuthorizer';
import type { AnyAttempt } from './authentication/lib/Attempt';
import { isValid } from './authentication/lib/Attempt';
import { BugButtonHandler } from './components/Button/BugButtonHandler';
import { ErrorBoundary } from './components/layout/ui/errors/error-boundary';
import { Modal, ModalRenderer } from './components/modal';
import { NetInfoDevOverlay } from './components/NetInfoDevOverlay';
import { NetInfoAutoToast } from './components/toast/net-info-auto-toast';
import { prepareAndDownloadTodaysIssue } from './download-edition/prepare-and-download-issue';
import { prepFileSystem } from './helpers/files';
import { nestProviders } from './helpers/provider';
import { pushDownloadFailsafe } from './helpers/push-download-failsafe';
import { isInBeta } from './helpers/release-stream';
import { newMobileProdStack } from './helpers/settings/defaults';
import { setApiUrl } from './helpers/settings/setters';
import { rollbackedApiUrlForBetaUsers } from './helpers/storage';
import { EditionProvider } from './hooks/use-edition-provider';
import { SettingsOverlayProvider } from './hooks/use-settings-overlay';
import { ToastProvider } from './hooks/use-toast';
import { pushNotificationRegistration } from './notifications/push-notifications';
import { DeprecateVersionModal } from './screens/deprecate-screen';
import { apolloClient } from './services/apollo-singleton';
import { errorService } from './services/errors';
import { loggingService } from './services/logging';
import { remoteConfigService } from './services/remote-config';

// Log Intitialisation
if (!__DEV__) {
	errorService.init(apolloClient);
}
loggingService.init(apolloClient);
remoteConfigService.init();

// --- SETUP OPERATIONS ---
pushNotificationRegistration();
prepFileSystem();

const styles = StyleSheet.create({
	appContainer: {
		flex: 1,
		backgroundColor: '#000',
	},
});

const persistenceKey = 'dev-nav-key-232asfdffgdfg1asdffgfdgfdga3413';

const isReactNavPersistenceError = (e: Error) =>
	__DEV__ && e.message.includes('There is no route defined for');

const WithProviders = nestProviders(
	Modal,
	ToastProvider,
	NavPositionProvider,
	ConfigProvider,
	EditionProvider,
	SettingsOverlayProvider,
);

const handleIdStatus = (attempt: AnyAttempt<IdentityAuthData>) =>
	setUserId(isValid(attempt) ? attempt.data.userDetails.id : null);

const shouldHavePushFailsafe = async (client: ApolloClient<object>) => {
	const largeRAM = await largeDeviceMemory();
	if (largeRAM) {
		pushDownloadFailsafe(client);
	}
};

const forceUpdateApiUrlIfBeta = async () => {
	// TODO - Remove this whole function before release to prod.
	// Previously we forced beta users to use a tmp url to point to the new stack, now that
	// we changed the fastly config we can rollback urls and eventually we can remove this full method.
	if (isInBeta()) {
		const rollbackedApi = await rollbackedApiUrlForBetaUsers.get();
		if (!rollbackedApi) {
			setApiUrl(apolloClient, newMobileProdStack);
			console.log('*** Beta: rolling back to default prod api url ***');
			await rollbackedApiUrlForBetaUsers.set(true);
		} else {
			console.log('*** Beta: api url for Beta users already updated ***');
		}
	}
};

export default class App extends React.Component {
	componentDidMount() {
		SplashScreen.hide();
		prepareAndDownloadTodaysIssue(apolloClient);
		shouldHavePushFailsafe(apolloClient);
		loggingService.postLogs();

		AppState.addEventListener('change', async (appState) => {
			if (appState === 'active') {
				prepareAndDownloadTodaysIssue(apolloClient);
				loggingService.postLogs();
				forceUpdateApiUrlIfBeta();
			}
		});

		{
			eventEmitter.on('editionCachesSet', () => {
				weatherHider(apolloClient);
			});
		}
	}

	async componentDidCatch(e: Error) {
		/**
		 * use an heuristic to check whether this is a react-nav error
		 * if it is then ditch our persistence and try to re-render
		 */
		if (isReactNavPersistenceError(e)) {
			await AsyncStorage.removeItem(persistenceKey);
			this.forceUpdate();
		}
	}
	/**
	 * When the component is mounted. This happens asynchronously and simply
	 * re-renders when we're good to go.
	 */
	render() {
		return (
			<ErrorBoundary>
				<ApolloProvider client={apolloClient}>
					<NetInfoDevOverlay>
						<WithProviders>
							<AccessProvider
								onIdentityStatusChange={handleIdStatus}
							>
								<StatusBar
									barStyle="light-content"
									backgroundColor="#041f4a"
								/>
								<View style={styles.appContainer}>
									<AppNavigation />
									<NetInfoAutoToast />
								</View>
								<ModalRenderer />
								<BugButtonHandler />
								<DeprecateVersionModal />
							</AccessProvider>
						</WithProviders>
					</NetInfoDevOverlay>
				</ApolloProvider>
			</ErrorBoundary>
		);
	}
}

/**
 * This needs to match what's found in your app_delegate.m and MainActivity.java.
 */

export const APP_NAME = 'Mallard';
