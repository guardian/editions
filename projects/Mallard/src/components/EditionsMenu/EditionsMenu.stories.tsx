import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { withKnobs } from '@storybook/addon-knobs'
import { EditionsMenu } from './EditionsMenu'
import { editions, Edition } from 'src/common'
import { SpecialEdition } from '../../../../Apps/common/src'

const props: { specialEditions: SpecialEdition[] } = {
    specialEditions: [
        {
            edition: 'daily-edition' as Edition,
            expiry: new Date(98, 1),
            buttonImageUri:
                'https://media.guim.co.uk/49cebb0db4a3e4d26d7d190da7be4a2e9bd7534f/0_0_103_158/103.png',
            topic: 'food',
            editionType: 'Special',
            notificationUTCOffset: 0,
            header: {
                title: `Food
Monthly`,
            },
            title: `Food
Monthly`,
            subTitle:
                'Store cupboard special: 20 quick and easy lockdown suppers',
            buttonStyle: {
                backgroundColor: '#FEEEF7',
                expiry: {
                    color: '#7D0068',
                    font: 'GuardianTextSans-Regular',
                    lineHeight: 16,
                    size: 15,
                },

                subTitle: {
                    color: '#7D0068',
                    font: 'GuardianTextSans-Bold',
                    lineHeight: 20,
                    size: 17,
                },
                title: {
                    color: '#121212',
                    font: 'GHGuardianHeadline-Regular',
                    lineHeight: 34,
                    size: 34,
                },
                image: {
                    height: 134,
                    width: 87,
                },
            },
        },
    ],
}

storiesOf('EditionsMenu', module)
    .addDecorator(withKnobs)
    .add('EditionsMenu - default', () => (
        <EditionsMenu
            navigationPress={() => {}}
            selectedEdition={editions.daily}
            storeSelectedEdition={() => {}}
        />
    ))
    .add('EditionsMenu - with Special Edition', () => (
        <EditionsMenu
            navigationPress={() => {}}
            selectedEdition={editions.daily}
            storeSelectedEdition={() => {}}
            {...props}
        />
    ))
