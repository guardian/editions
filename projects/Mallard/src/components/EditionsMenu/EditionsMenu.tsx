import React from 'react'
import { FlatList, ScrollView } from 'react-native'
import { EditionId, RegionalEdition, SpecialEdition } from 'src/common'
import { StoreSelectedEditionFunc } from 'src/hooks/use-edition-provider'
import { defaultRegionalEditions } from '../../../../Apps/common/src/editions-defaults'
import { EditionsMenuHeader } from './Header/Header'
import { ItemSeperator } from './ItemSeperator/ItemSeperator'
import { RegionButton } from './RegionButton/RegionButton'
import { SpecialEditionButton } from './SpecialEditionButton/SpecialEditionButton'

const EditionsMenu = ({
    navigationPress,
    regionalEditions,
    selectedEdition,
    specialEditions,
    storeSelectedEdition,
}: {
    navigationPress: () => void
    regionalEditions?: RegionalEdition[]
    selectedEdition: EditionId
    specialEditions?: SpecialEdition[]
    storeSelectedEdition: StoreSelectedEditionFunc
}) => {
    return (
        <ScrollView>
            <EditionsMenuHeader>Regions</EditionsMenuHeader>
            <FlatList
                data={regionalEditions || defaultRegionalEditions}
                renderItem={({ item }: { item: RegionalEdition }) => {
                    return (
                        <RegionButton
                            selected={
                                selectedEdition === item.edition ? true : false
                            }
                            onPress={() => {
                                storeSelectedEdition(item, 'RegionalEdition')
                                navigationPress()
                            }}
                            title={item.title}
                            subTitle={item.subTitle}
                        />
                    )
                }}
                ItemSeparatorComponent={() => <ItemSeperator />}
            />
            {specialEditions && specialEditions.length > 0 && (
                <>
                    <EditionsMenuHeader>Special Editions</EditionsMenuHeader>
                    <FlatList
                        data={specialEditions}
                        renderItem={({ item }: { item: SpecialEdition }) => {
                            const {
                                buttonStyle,
                                buttonImageUri,
                                edition,
                                expiry,
                                title,
                                subTitle,
                            } = item
                            return (
                                <SpecialEditionButton
                                    buttonImageUri={buttonImageUri}
                                    expiry={expiry}
                                    onPress={() => {
                                        storeSelectedEdition(
                                            item,
                                            'SpecialEdition',
                                        )
                                        navigationPress()
                                    }}
                                    title={title}
                                    selected={
                                        selectedEdition === edition
                                            ? true
                                            : false
                                    }
                                    style={buttonStyle}
                                    subTitle={subTitle}
                                />
                            )
                        }}
                    />
                </>
            )}
        </ScrollView>
    )
}

export { EditionsMenu }
