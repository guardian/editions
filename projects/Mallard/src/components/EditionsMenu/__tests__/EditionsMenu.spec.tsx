import React from 'react'
import TestRenderer, { ReactTestRendererJSON } from 'react-test-renderer'
import { Edition, editions } from 'src/common'
import { EditionsMenu } from '../EditionsMenu'

jest.mock('src/components/EditionsMenu/RegionButton/RegionButton', () => ({
    RegionButton: () => 'RegionButton',
}))

jest.mock(
    'src/components/EditionsMenu/SpecialEditionButton/SpecialEditionButton',
    () => ({
        SpecialEditionButton: () => 'SpecialEditionButton',
    }),
)

jest.mock('src/components/EditionsMenu/Header/Header', () => ({
    EditionsMenuHeader: () => 'EditionsMenuHeader',
}))

const regionalEditions = [
    {
        title: 'The UK Daily Edition',
        subTitle: 'Published every day by 12am (GMT)',
        edition: editions.daily as Edition,
        header: {
            title: 'The Daily',
        },
    },
    {
        title: 'Australia Daily',
        subTitle: 'Published every day by 9:30am (AEST)',
        edition: editions.ausWeekly as Edition,
        header: {
            title: 'Austraila',
            subTitle: 'Weekender',
        },
    },
    {
        title: 'US and Cananda Weekend',
        subTitle: 'Published every Saturday by 8am (EST)',
        edition: editions.usWeekly as Edition,
        header: {
            title: 'US',
            subTitle: 'Weekender',
        },
    },
]

const specialEditions = [
    {
        edition: 'daily-edition' as Edition,
        expiry: new Date(98, 1),
        image: {
            source: 'media',
            path: '/path/to/image',
        },
        title: `Food
Monthly`,
        subTitle: 'Store cupboard special: 20 quick and easy lockdown suppers',
        header: {
            title: 'Food',
            subTitle: 'Monthly',
        },
        buttonStyle: {
            backgroundColor: '#FEEEF7',
            expiry: {
                color: '#7D0068',
                font: 'GuardianTextSans-Regular',
                lineHeight: 16,
                size: 15,
            },

            subTitle: {
                color: '#7D0068',
                font: 'GuardianTextSans-Bold',
                lineHeight: 20,
                size: 17,
            },
            title: {
                color: '#121212',
                font: 'GHGuardianHeadline-Regular',
                lineHeight: 34,
                size: 34,
            },
            image: {
                height: 134,
                width: 87,
            },
        },
    },
]

const props = {
    navigationPress: () => {},
    selectedEdition: editions.daily,
    storeSelectedEdition: () => {},
}

describe('EditionsMenu', () => {
    it('should display a default EditionsMenu with correct styling and default Regional Buttons', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <EditionsMenu {...props} />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
    it('should display a EditionsMenu with correct styling and alternative Regional Buttons', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <EditionsMenu {...props} regionalEditions={regionalEditions} />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
    it('should display a EditionsMenu with correct styling default Regional Buttons and a Special Edition Button', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <EditionsMenu {...props} specialEditions={specialEditions} />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
})
