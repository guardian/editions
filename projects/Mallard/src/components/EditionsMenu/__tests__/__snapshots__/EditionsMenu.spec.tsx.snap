// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditionsMenu should display a EditionsMenu with correct styling and alternative Regional Buttons 1`] = `
<RCTScrollView>
  <View>
    EditionsMenuHeader
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "edition": "daily-edition",
            "editionType": "Regional",
            "header": Object {
              "title": "The Daily",
            },
            "notificationUTCOffset": 1,
            "subTitle": "Published every day by 12am (GMT)",
            "title": "The UK Daily Edition",
            "topic": "au",
          },
          Object {
            "edition": "australian-edition",
            "editionType": "Regional",
            "header": Object {
              "subTitle": "Weekender",
              "title": "Austraila",
            },
            "notificationUTCOffset": 1,
            "subTitle": "Published every day by 9:30am (AEST)",
            "title": "Australia Daily",
            "topic": "au",
          },
          Object {
            "edition": "american-edition",
            "editionType": "Regional",
            "header": Object {
              "subTitle": "Weekender",
              "title": "US",
            },
            "notificationUTCOffset": 1,
            "subTitle": "Published every Saturday by 8am (EST)",
            "title": "US and Cananda Weekend",
            "topic": "au",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          RegionButton
          <View
            style={
              Object {
                "backgroundColor": "#121212",
                "height": 1,
              }
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          RegionButton
          <View
            style={
              Object {
                "backgroundColor": "#121212",
                "height": 1,
              }
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          RegionButton
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTScrollView>
`;

exports[`EditionsMenu should display a EditionsMenu with correct styling default Regional Buttons and a Special Edition Button 1`] = `
<RCTScrollView>
  <View>
    EditionsMenuHeader
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "edition": "daily-edition",
            "editionType": "Regional",
            "header": Object {
              "title": "The Daily",
            },
            "notificationUTCOffset": 3,
            "subTitle": "Published every morning
by 6am (GMT)",
            "title": "The Daily",
            "topic": "uk",
          },
          Object {
            "edition": "australian-edition",
            "editionType": "Regional",
            "header": Object {
              "subTitle": "Weekender",
              "title": "Australia",
            },
            "notificationUTCOffset": -5,
            "subTitle": "Published every Saturday morning
by 6am (AEST)",
            "title": "Australia Weekender",
            "topic": "au",
          },
          Object {
            "edition": "american-edition",
            "editionType": "Regional",
            "header": Object {
              "subTitle": "Weekender",
              "title": "US",
            },
            "notificationUTCOffset": 8,
            "subTitle": "Published every Saturday morning
by 6am (EST)",
            "title": "US Weekender",
            "topic": "us",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          RegionButton
          <View
            style={
              Object {
                "backgroundColor": "#121212",
                "height": 1,
              }
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          RegionButton
          <View
            style={
              Object {
                "backgroundColor": "#121212",
                "height": 1,
              }
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          RegionButton
        </View>
      </View>
    </RCTScrollView>
    EditionsMenuHeader
    <RCTScrollView
      data={
        Array [
          Object {
            "buttonStyle": Object {
              "backgroundColor": "#FEEEF7",
              "expiry": Object {
                "color": "#7D0068",
                "font": "GuardianTextSans-Regular",
                "lineHeight": 16,
                "size": 15,
              },
              "image": Object {
                "height": 134,
                "width": 87,
              },
              "subTitle": Object {
                "color": "#7D0068",
                "font": "GuardianTextSans-Bold",
                "lineHeight": 20,
                "size": 17,
              },
              "title": Object {
                "color": "#121212",
                "font": "GHGuardianHeadline-Regular",
                "lineHeight": 34,
                "size": 34,
              },
            },
            "edition": "daily-edition",
            "editionType": "Special",
            "expiry": 1998-02-01T00:00:00.000Z,
            "header": Object {
              "subTitle": "Monthly",
              "title": "Food",
            },
            "image": Object {
              "path": "/path/to/image",
              "source": "media",
            },
            "notificationUTCOffset": 1,
            "subTitle": "Store cupboard special: 20 quick and easy lockdown suppers",
            "title": "Food
Monthly",
            "topic": "food",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          SpecialEditionButton
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTScrollView>
`;

exports[`EditionsMenu should display a default EditionsMenu with correct styling and default Regional Buttons 1`] = `
<RCTScrollView>
  <View>
    EditionsMenuHeader
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "edition": "daily-edition",
            "editionType": "Regional",
            "header": Object {
              "title": "The Daily",
            },
            "notificationUTCOffset": 3,
            "subTitle": "Published every morning
by 6am (GMT)",
            "title": "The Daily",
            "topic": "uk",
          },
          Object {
            "edition": "australian-edition",
            "editionType": "Regional",
            "header": Object {
              "subTitle": "Weekender",
              "title": "Australia",
            },
            "notificationUTCOffset": -5,
            "subTitle": "Published every Saturday morning
by 6am (AEST)",
            "title": "Australia Weekender",
            "topic": "au",
          },
          Object {
            "edition": "american-edition",
            "editionType": "Regional",
            "header": Object {
              "subTitle": "Weekender",
              "title": "US",
            },
            "notificationUTCOffset": 8,
            "subTitle": "Published every Saturday morning
by 6am (EST)",
            "title": "US Weekender",
            "topic": "us",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          RegionButton
          <View
            style={
              Object {
                "backgroundColor": "#121212",
                "height": 1,
              }
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          RegionButton
          <View
            style={
              Object {
                "backgroundColor": "#121212",
                "height": 1,
              }
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          RegionButton
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTScrollView>
`;
